lib/dataset/bins.cpp

```diff
@@ -211,6 +212,18 @@ Variable lookup_previous(const DataArray &function, const Variable &x, Dim dim,
                              "lookup_previous");
 }
 
+Variable pretend_bins_for_threading(const DataArray &da, Dim bin_dim) {
+  const auto dim = da.dims().inner();
+  const auto size = std::max(scipp::index(1), da.dims()[dim]);
+  // TODO automatic setup with reasonable bin count
```

Comment:
Would doing this help with threading? What is missing to do it?

Reply:
Coming up with a smart way determining a bin count, instead of using a hard-coded number with common factors of core counts?